functions/config-versioning.bash

```diff
@@ -0,0 +1,310 @@
+            return 0
+          fi
+      return 1
+    fi
+  else
+    echo "NO - Create new"
+    cond_redirect git init / 
+    git -C "/" remote add origin "$repourl"
+    echo "$logPrefix This file is intentionly left empty" > /openhabian_config_versioning
+    cp "${BASEDIR:-/opt/openhabian}/includes/cfg-gitignore.txt" /.gitignore
+  fi
+
+  if ! git -C "/" config user.name
+  then 
+    git -C "/" config user.user "$user"
+  else 
+    exsisting_userename=$(git -c "/" config user.name)
+    echo "$logPrefix Git name already set [$exsisting_userename]... SKIPPED"
+  fi
+
+  if ! git -C "/" config user.email
+  then 
+    git -C "/" config user.email "$email"
+  else 
+    exsisting_email=$(git -c "/" config user.email)
+    echo "$logPrefix Git email already set [$exsisting_email]... SKIPPED"
+  fi
+
+  git -C "/" config versioning.repositiory "$repositiory"
+  git -C "/" config versioning.token "$token"
+  cond_redirect git -C "/" push --set-upstream origin master
+  git -C "/" config alias.cmp '!f() { git add -A && git commit -m "$@" && git push; }; f'
+
+  if [[ -n $INTERACTIVE ]]
+  then
+    systemctl start openhab.service
+    whiptail --title "Operation Successful!" --msgbox "Setup was successful.\\n\\nGithub repository with the name [$repositiory] is now up and running.\\nIt is ready for you to commit.\\n\\nYou can also enable auto commit daily" 15 80
+  fi
+
+}
+
+
+## Function for uninstall up github repository for config versioning.
+## This function can be invoked only INTERACTIVE.
+##
+##    versioning_uninstall()
+##
+versioning_uninstall() {
+
+  if [ ! -f /.gitignore ] && [ ! -d /.git ]
+  then
+    echo "$(timestamp) [openHABian] No Versioning installation found "
+    (whiptail --title "Versioning Github" --yes-button "Continue" --no-button "Cancel" --yesno "No Versioning Found" 14 80) ; return 0
+  fi
+  currentrepo=$(cond_redirect sed -n -e "s/^.*\\(branch \\)/\\1/p" /.git/FETCH_HEAD)
+  if (whiptail --title "Versioning Github" --yes-button "Continue" --no-button "Cancel" --yesno "Found Existing Repositiory\\n\\n$currentrepo\\n\\nUninstall Versioning are you sure you want to continue" 14 80)
+  then
+    echo "OK"
+  else
+    return 0
+  fi
+
+  echo -n "$(timestamp) [openHABian] Uninstalling Git Versioning..."
+  if cond_redirect rm -rf /.git; then echo -n ".OK"; else echo ".FAILED (remove .git)"; return 1; fi
+  if cond_redirect rm /.gitignore; then echo -n ".OK"; else echo ".FAILED (remove .gitignore)"; return 1; fi
+  if cond_redirect rm /openhabian_config_versioning; then echo -n ".OK"; else echo ".FAILED (remove install check)"; return 1; fi
+  echo ".Successfull"
+
+  if [[ -n $INTERACTIVE ]]
+  then
+    whiptail --title "Operation Successfull" --msgbox "Local Versioning Removed" 5 80
+  fi
+}
+
+## Enable timer to run commit.
+## This function INTERACTIVE
+## 
+##    versioning_enable_auto_commit()
+##
+versioning_enable_auto_commit() {
```

Comment:
err ... ok it's hard to understand your programming logic when reading code in 'diff' form.

Could you spend some sentences to explain it ?